apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -337,3 +355,25 @@ max_payload_size(
     PayloadSize;
 max_payload_size(_DNSMessage) ->
     ?MAX_PACKET_SIZE.
+
+%%%===================================================================
+%%% Metrics functions
+%%%===================================================================
+
+-spec(init_metrics() -> ok).
+init_metrics() ->
+    prometheus_counter:new([{name, dns_forwarder_failures_total},
+                            {labels, [zone, reason]},
+                            {help, "Number of failed DNS queries forwarded to `forwarder` at `zone`"}]),
+    prometheus_histogram:new([{name, dns_forwarder_requests_duration_seconds},
+                              {labels, [zone]},
+                              {duration_unit, false},
```

Comment:
You don't need `report_query_response_time/2` function if `duration_unit` is set to `seconds`. The code would be much cleaner in that case, e.g. see https://github.com/dcos/lashup/commit/d46a789b34ef915fcfc81b44e21a49fcd1396725